### DISCUSSION TITLE
Is it possible to prevent combobox input to focus when clicking combobox item?

### DISCUSSION THREAD

**Discussion by sclebal** [ariakit/ariakit#3945] [Questions] [answered]
I noticed in this example https://ariakit.org/examples/select-combobox, that if you unfocus the `Ariakit.Combobox` by clicking somewhere in the popover, the combobox gets the focus back when pointerdown event is triggered in one of the select items (same happens with combobox items).

https://github.com/user-attachments/assets/ddbe9816-d127-40f6-bad2-3fa10b020069

_You can notice that in 00:01 the combobox input gets focused by clicking an item_

I wonder if it would be possible to disable that behavior, it feels a bit odd that when you're selecting something, the focus outline flashes for a second.

Thanks for your help 🙏🏻

**Comment by diegohaz** [comment, marked as answer]
When the [`virtualFocus`](https://ariakit.org/reference/combobox-provider#virtualfocus) prop is enabled, the combobox should receive focus when an item is focused (e.g., on pointer down).

You can apply a CSS transition with a delay on the input outline to ease this state change.

You can also completely disable the `virtualFocus` prop, but this will also disable the [`autoSelect`](https://ariakit.org/reference/combobox#autoselect) feature.

Alternatively, you can disable `virtualFocus` only when the input loses focus:

```jsx
const [virtualFocus, setVirtualFocus] = useState(true);

<Ariakit.ComboboxProvider
  virtualFocus={virtualFocus}
  setOpen={() => setVirtualFocus(true)}
>
  ...
  <Ariakit.Combobox
    onFocus={() => setVirtualFocus(true)}
    onBlur={() => setVirtualFocus(false)}
  />
```
